app_dart/lib/src/request_handlers/luci_status.dart

```diff
@@ -105,14 +112,13 @@ class LuciStatusHandler extends RequestHandler<Body> {
       switch (build.failureReason) {
         case FailureReason.buildbucketFailure:
         case FailureReason.infraFailure:
-          // infra failed
-          await _rescheduleBuild(
-            sha: sha,
+          await luciBuildService.rescheduleBuild(
+            commitSha: sha,
             builderName: builderName,
             build: build,
             retries: retries,
           );
-          return;
+          break;
```

Comment:
Good point, reschedule now returns a boolean if it was rescheduled then it will set the status to pending. Otherwise will set the status to complete and failure.